pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -0,0 +1,141 @@
+package bucketaccessrequest
+
+import (
+	"context"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kubernetes-sigs/container-object-storage-interface-api/apis/objectstorage.k8s.io/v1alpha1"
+	bucketclientset "github.com/kubernetes-sigs/container-object-storage-interface-api/clientset"
+	bucketcontroller "github.com/kubernetes-sigs/container-object-storage-interface-api/controller"
+	"github.com/kubernetes-sigs/container-object-storage-interface-controller/pkg/util"
+	kubeclientset "k8s.io/client-go/kubernetes"
+
+	"github.com/golang/glog"
+)
+
+type bucketAccessRequestListener struct {
+	kubeClient   kubeclientset.Interface
+	bucketClient bucketclientset.Interface
+}
+
+func NewListener() bucketcontroller.BucketAccessRequestListener {
+	return &bucketAccessRequestListener{}
+}
+
+func (b *bucketAccessRequestListener) InitializeKubeClient(k kubeclientset.Interface) {
+	b.kubeClient = k
+}
+
+func (b *bucketAccessRequestListener) InitializeBucketClient(bc bucketclientset.Interface) {
+	b.bucketClient = bc
+}
+
+func (b *bucketAccessRequestListener) Add(ctx context.Context, obj *v1alpha1.BucketAccessRequest) error {
+	glog.V(1).Infof("add called for bucketaccessrequest %s", obj.Name)
+	bucketAccessRequest := obj
+
+	err := b.provisionBucketAccess(ctx, bucketAccessRequest)
+	if err != nil {
+		// Provisioning is 100% finished / not in progress.
+		switch err {
+		case util.ErrInvalidBucketAccessClass:
+			glog.V(5).Infof("Bucket access class specified does not exist. Stop provisioning, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.UID)
+			err = nil
+		case util.ErrBucketAccessAlreadyExists:
+			glog.V(5).Infof("Bucket access already exist for this bucket request. Stop provisioning, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.UID)
```

Comment:
"removing bucket access request %s from bucket access requests in progress" is not understandable to me.

Reply:
lack of better wording... I am trying to say
"Bucket access already exist for this bucket request. Stop provisioning, removing bucket access request br1 from bucket access requests in progress"
can also be changed to 
"Bucket access already exist for this bucket request br1."

Reply:
Fixed the wording